cache/src/main/java/io/envoyproxy/controlplane/cache/SimpleCache.java

```diff
@@ -458,21 +463,35 @@ private boolean respond(Watch watch, U snapshot, T group) {
     Map<String, VersionedResource<?>> snapshotResources =
         snapshot.versionedResources(watch.request().getResourceType());
 
+    boolean allowDefaultResource = false;
     if (!watch.request().getResourceNamesList().isEmpty() && watch.ads()) {
       Collection<String> missingNames = watch.request().getResourceNamesList().stream()
           .filter(name -> !snapshotResources.containsKey(name))
           .collect(Collectors.toList());
 
       if (!missingNames.isEmpty()) {
-        LOGGER.info(
-            "not responding in ADS mode for {} from node {} at version {} for request [{}] since [{}] not in snapshot",
-            watch.request().getTypeUrl(),
-            group,
-            snapshot.version(watch.request().getResourceType(), watch.request().getResourceNamesList()),
-            String.join(", ", watch.request().getResourceNamesList()),
-            String.join(", ", missingNames));
+        if (watch.allowIncompleteEdsUpdate() && watch.request().getResourceType().equals(ResourceType.ENDPOINT)) {
+          allowDefaultResource = true;
+          LOGGER.info(
+              "responding incomplete EDS update in ADS mode for {} from node {} at version {} for request [{}] "
```

Comment:
nit: "responding with incomplete"

Reply:
minor: should this be debug since in this mode, technically incomplete EDS update is allowed

Reply:
I used `info` to be consistent with the scenario below when we ignore incomplete ADS which is also allowed

it is allowed, but should not happen often in theory.